src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -225,20 +228,33 @@ export class FocusTrapZone extends React.Component<FocusTrapZoneProps, {}> {
 
   private _handleOutsideClick = (ev: MouseEvent): void => {
     const clickedElement = ev.target as HTMLElement
-    clickedElement && this._forceFocusInTrap(ev, clickedElement)
+    const { isClickableOutsideFocusTrap, focusTriggerOnOutsideClick } = this.props
+
+    if (!isClickableOutsideFocusTrap) {
+      clickedElement && this._forceFocusInTrap(ev, clickedElement)
+    } else if (!focusTriggerOnOutsideClick) {
+      const isOutsideFocusTrapZone =
+        this._root.current && !this._root.current.contains(clickedElement)
+      const isOutsideTriggerElement =
+        this._previouslyFocusedElementOutsideTrapZone &&
+        !this._previouslyFocusedElementOutsideTrapZone.contains(clickedElement)
+      if (isOutsideFocusTrapZone && isOutsideTriggerElement) {
+        // set it to NULL, so the trigger will not be focused on componentWillUnmount
+        this._previouslyFocusedElementOutsideTrapZone = null
```

Comment:
here we are implicitly assume that the element that was previously focused is a trigger element - while, in general case, it might be not the case. The problem I see is that this implicit assumption is made as part of the FocusZone API - in other words, this introduced prop targets specific set up and, in general case, might not work as expected (suppose that element other than `_previouslyFocusedElementOutsideTrapZone` has triggered focus zone - in that case `focusTriggerOnOutsideClick` will not work as expected, as `_previouslyFocusedElementOutsideTrapZone` !== `triggerElement`)

Reply:
you're right, it can be not always an element which triggered focus zone, if a user sets the prop `elementToFocusOnDismiss`, it will replace the trigger element. That seems the only case.
To be honest I have no idea how to name it differently